pysr/regressor_sequence.py

```diff
@@ -0,0 +1,381 @@
+        X_units: Optional[ArrayLike[str]] = None,
+        y_units=None,
+    ) -> "PySRSequenceRegressor":
+        """
+        Search for equations to fit the time series dataset and store them in `self.equations_`.
+
+        Parameters
+        ----------
+        X : ndarray | pandas.DataFrame
+            Time series training data of shape (n_times, n_features).
+        weights : ndarray | pandas.DataFrame
+            Weight array of the same shape as `X`.
+            Each element is how to weight the mean-square-error loss
+            for that particular element of `X`. Alternatively,
+            if a custom `loss` was set, it will can be used
+            in arbitrary ways.
+        variable_names : list[str]
+            A list of names for the variables, rather than "x0t_1", "x1t_2", etc.
+            If `X` is a pandas dataframe, the column name will be used
+            instead of `variable_names`. Cannot contain spaces or special
+            characters. Avoid variable names which are also
+            function names in `sympy`, such as "N".
+            The number of variable names must be equal to (n_features,).
+        X_units : list[str]
+            A list of units for each variable in `X`. Each unit should be
+            a string representing a Julia expression. See DynamicQuantities.jl
+            https://symbolicml.org/DynamicQuantities.jl/dev/units/ for more
+            information.
+            Length should be equal to n_features.
+
+        Returns
+        -------
+        self : object
+            Fitted estimator.
+        """
+        X = self._validate_data(X)
+        _check_assertions(
+            X,
+            y,
+            Xresampled,
+            self.recursive_history_length,
+            weights,
+            variable_names,
+            X_units,
+            y_units,
+        )
+
+        current_X = X[self.recursive_history_length :]
+        historical_X = np.lib.stride_tricks.sliding_window_view(
+            X[:-1].flatten(), self.recursive_history_length * X.shape[1]
+        )[:: current_X.shape[1], :]
+        y_units = X_units
+        if isinstance(weights, np.ndarray):
+            weights = weights[self.recursive_history_length :]
+        variable_names = self._construct_variable_names(
+            current_X.shape[1], variable_names
+        )
+
+        self._regressor.fit(
+            X=historical_X,
+            y=current_X,
+            weights=weights,
+            variable_names=variable_names,
+            X_units=X_units,
+            y_units=y_units,
+            complexity_of_variables=complexity_of_variables,
+        )
+        self._regressor.__dict__["__sklearn_is_fitted__"] = True
+        self._regressor.__dict__["selection_mask_"] = self._regressor.selection_mask_
+        self._regressor.__dict__["feature_names_in_"] = (
+            self._regressor.feature_names_in_
+        )
+        self._regressor.__dict__["nout_"] = self._regressor.nout_
```

Comment:
Why do these use `__dict__` here instead of other ways of setting these?

Reply:
And also why are each of these needed? I think some of them are internal properties of `PySRRegressor`. But if they're not needed within the sequence regressor there's likely no reason to include them.